ui/src/tool-panel/switch/dialog/switch-draft-oid-field.tsx

```diff
@@ -81,82 +116,110 @@ export const SwitchDraftOidField: React.FC<SwitchDraftOidFieldProps> = ({
-                      : undefined,
-
-                  existingInGeoviite?.stateCategory &&
-                  existingInGeoviite.stateCategory !== 'NOT_EXISTING'
-                      ? t('switch-dialog.oid-in-use')
-                      : undefined,
               ].filter(filterNotEmpty)
             : [];
 
     const mandatoryFieldError =
-        isVisited && draftOidFieldOpen && draftOid.trim().length === 0
+        isVisited && editingOid && draftOid.trim().length === 0
             ? [t('switch-dialog.mandatory-field')]
             : [];
 
     const combinedErrors = [...errors, ...mandatoryFieldError, ...oidPresenceErrors];
 
+    const oidPresenceWarnings =
+        editingOid && !loadingOidPresence && oidPresence && oidIsLocallyValid
+            ? [
+                  existingInGeoviite?.stateCategory &&
+                  existingInGeoviite.stateCategory === 'NOT_EXISTING' ? (
+                      <React.Fragment>
+                          <div>{t('switch-dialog.oid-in-use-deleted')}</div>
+                          <div>
+                              <ExistingSwitchEditLink
+                                  onEdit={onEdit}
+                                  setDraftOid={setDraftOid}
+                                  setDraftOidFieldOpen={setEditingOid}
+                                  existingInGeoviite={existingInGeoviite}
+                              />
+                          </div>
+                      </React.Fragment>
+                  ) : undefined,
+
+                  existingInGeoviite?.stateCategory &&
+                  existingInGeoviite.stateCategory !== 'NOT_EXISTING' ? (
+                      <React.Fragment>
+                          <div>{t('switch-dialog.oid-in-use')}</div>
+                          <div>
+                              <ExistingSwitchEditLink
+                                  onEdit={onEdit}
+                                  setDraftOid={setDraftOid}
+                                  setDraftOidFieldOpen={setEditingOid}
+                                  existingInGeoviite={existingInGeoviite}
+                              />
+                          </div>
+                      </React.Fragment>
+                  ) : undefined,
+              ].filter(filterNotEmpty)
+            : [];
+
     const oidOk =
         !loadingOidPresence &&
         combinedErrors.length === 0 &&
         draftOid.length > 0 &&
         oidPresence?.existsInRatko === true;
 
+    const spinnerIfLoading =
+        editingOid && loadingOidPresence ? <Spinner size={SpinnerSize.SMALL} inline /> : undefined;
+    const oidOkIndicator = oidOk && (
+        <span className={styles['switch-edit-dialog__switch-oid-ok']}>
+            <Icons.Tick size={IconSize.SMALL} color={IconColor.INHERIT} />
+            {t('switch-dialog.oid-was-found-from-ratko')}
+        </span>
+    );
+
+    const helpComponent = spinnerIfLoading ?? oidOkIndicator;
```

Comment:
Latausspinneri ja OK-indikaatio ovat nyt valinnaisia, ja molemmat näytetään `FieldLayout`:in help-osiossa